### PR TITLE
save reference frame numbers to Dataset for Clustering

### PR DESCRIPTION
**Pull request by hainm**
Related to #177

I am trying to do easy part.

```python
In [3]: pt.cluster.kmeans(traj, n_clusters=5, dtype='dataset')
Out[3]:
<pytraj.DatasetList with 2 datasets>
refframe
[4 0 8 6 7]

cluster_number
[1 1 0 0 0 0 3 4 2 2]
```

### CODE REVIEW

File: src/Analysis_Clustering.cpp [hunk 1]
```diff
@@ -228,6 +229,10 @@ Analysis::RetType Analysis_Clustering::Setup(ArgList& analyzeArgs, DataSetList*
   // Output option for cluster info
   suppressInfo_ = analyzeArgs.hasKey("noinfo");
 
+  // Dataset to store representative cluster frames
+  clustersRefFrame_ = (DataSet_integer*) datasetlist->AddSet(DataSet::INTEGER, "refframe", "refframe");
+  if (clustersRefFrame_==0) return Analysis::ERR;
+
   // Dataset to store cluster number v time
```
Comment (drroe) [inline src/Analysis_Clustering.cpp]: This should come after the cluster num vs time data set is allocated and then use that DataSet name. All DataSets generated by an Action or an Analysis should have the same name. In this case you want to use the name of the `cnumvtime_` DataSet and add the Aspect 'refframe', e.g.
```C++
clustersRefFrame_ = datasetlist->AddSet(DataSet::Integer, MetaData(cnumvtime_->Meta().Name(), "refframe"));
```
Also, you don't need to make `clustersRefFrame_` a pointer to DataSet_integer. Just make it a `DataSet*`. Then you avoid having the `DataSet_integer.h` include in `Analysis_Clustering.h` and make things a bit more general. You can always cast to DataSet_integer inside `Analysis_Clustering.cpp` if need be.

Comment (hainm) [inline src/Analysis_Clustering.cpp]: Can you make another PR for this? I will close this one rather pushing new commit. It is easier for both of us. :d

Hai

Comment (hainm) [inline src/Analysis_Clustering.cpp]: Ps: i will keep in mind about aspect and casting dataset.

**Comment by drroe** [comment]
> Can you make another PR for this? I will close this one rather pushing new commit. It is easier for both of us. :d

I'm delaying on this a little bit because I'm wondering if all clustering data shouldn't be stored in either one specialized DataSet or in multiple DataSets with different aspects. Former would allow cluster output to be used with the DataFile framework more easily, latter would be easier to implement in the short run but maybe harder to maintain. Maybe this should be a separate issue? Not sure...

**Pull request closed without merging**